server/src/main/java/com/objectcomputing/checkins/services/pulseresponse/PulseResponseController.java

```diff
@@ -1,67 +1,105 @@
+                .body(error);
+    }
 
     /**
      * Find Pulse Response by Team Member or Date Range.
      * 
+     * @param teamMemberId
      * @param dateFrom
      * @param dateTo
-     * @param teamMemberId
      * @return
      */
-    @Get("/{?dateFrom,dateTo,teamMemberId}")
-    public HttpResponse<List<PulseResponse>> findByValue(@Nullable @Format("yyyy-MM-dd") LocalDate dateFrom,
-            @Nullable @Format("yyyy-MM-dd") LocalDate dateTo, @Nullable UUID teamMemberId) {
+    @Get("/{?teamMemberId,dateFrom,dateTo}")
+    public Set<PulseResponse> findByValue(@Nullable @Format("yyyy-MM-dd") LocalDate dateFrom, @Nullable @Format("yyyy-MM-dd") LocalDate dateTo,@Nullable UUID teamMemberId) {          
+            return pulseResponseServices.findByFields(teamMemberId, dateFrom, dateTo);
+    }
+
+    /**
+     * Create and save a new PulseResponse.
+     *
+     * @param pulseResponse, {@link PulseResponseCreateDTO}
+     * @return {@link HttpResponse<PulseResponse>}
+     */
 
-        return pulseResponseService.findBy(dateFrom, dateTo, teamMemberId);
+    @Post(value = "/")
+    public HttpResponse<PulseResponse> createAPulseResponse(@Body @Valid PulseResponseCreateDTO pulseResponse,
+                                                                HttpRequest<PulseResponseCreateDTO> request) {
+        PulseResponse newMemberPulseResponse = pulseResponseServices.save(new PulseResponse(pulseResponse.getSubmissionDate(),pulseResponse.getUpdatedDate(), pulseResponse.getTeamMemberId(), pulseResponse.getInternalFeelings(), pulseResponse.getExternalFeelings()));
+        return HttpResponse
+                .created(newMemberPulseResponse)
+                .headers(headers -> headers.location(URI.create(String.format("%s/%s", request.getUri(), newMemberPulseResponse.getId()))));
     }
 
     /**
-     * Save a new Pulse Response.
+     * Update pulseresponse details
      * @param pulseResponse
      * @return
      */
-    @Post("/")
-    public HttpResponse<PulseResponse> save(@Body @Valid PulseResponse pulseResponse) {
-        return pulseResponseService.save(pulseResponse);
+    @Put("/")
+    public HttpResponse<?> update(@Body @Valid PulseResponse pulseResponse,HttpRequest<PulseResponseCreateDTO> request) {
+
+            PulseResponse updatedMemberPulseResponse = pulseResponseServices.update(pulseResponse);
+            return HttpResponse
+                    .ok()
+                    .headers(headers -> headers.location(URI.create(String.format("%s/%s", request.getPath(),updatedMemberPulseResponse.getId()))))
+                    .body(updatedMemberPulseResponse);                  
     }
 
     /**
-     * Update a Pulse Response.
-     * @param pulseResponse
+     * 
+     * @param id
      * @return
      */
-    @Put("/")
-    public HttpResponse<PulseResponse> update(@Body @Valid PulseResponse pulseResponse) {
-        return pulseResponseService.update(pulseResponse);
+    @Get("/{id}")
```

Comment:
@mkimberlin Do we need this endpoint? It looks like this was not part of the original ticket - https://github.com/oci-labs/check-ins/issues/86